### ISSUE TITLE
No audio after Video Resample

### ISSUE THREAD

**Issue by HardikSuba** [hoolrory/AndroidVideoSamples#1]
There is no audio in generated video file.
I have used this code,

```
VideoResampler resampler = new VideoResampler();
        resampler.addSamplerClip(new SamplerClip(inputUri));
        resampler.setOutput(outputUri);
        resampler.setOutputResolution(Resolution.RESOLUTION_360P.getWidth(), Resolution.RESOLUTION_360P.getHeight());
        try {
            resampler.start();
        } catch (Throwable e) {
            e.printStackTrace();
        }
```

**Comment by microgomi** [comment]
Hi HardikSuba, I have a same problem,  did you resolve this issue?

**Comment by HardikSuba** [comment]
Hi microgomi, I am not able to solve it yet.

**Comment by gouravd** [comment]
I do not think this can be solved directly. You might want to mux the video from this converted file and audio from the original file together

**Comment by kaplandroid** [comment]
Any sample code for mux original audio?

**Comment by Berreis** [comment]
I am also having this problem? Does anyone have an idea on how to solve that?

**Comment by kaplandroid** [comment]
I added following  code to my application. "audioFile" is original video file.  You can try this.

> import com.coremedia.iso.boxes.Container;
> import com.googlecode.mp4parser.authoring.Movie;
> import com.googlecode.mp4parser.authoring.builder.DefaultMp4Builder;
> import com.googlecode.mp4parser.authoring.container.mp4.MovieCreator;
> 
> ```
> public Uri mux(String audioFile, String videoFile) {
>     String outputFilePath = videoFile.replace(".mp4", "_with_audio.mp4");
>     Uri outputFile = Uri.parse(outputFilePath);
>     Movie video;
> 
>     try {
>         video = new MovieCreator().build(videoFile);
>     } catch (RuntimeException e) {
>         e.printStackTrace();
>         return null;
>     } catch (IOException e) {
>         e.printStackTrace();
>         return null;
>     }
> 
> 
>     Movie audio;
>     try {
>         audio = new MovieCreator().build(audioFile);
>     } catch (IOException e) {
>         e.printStackTrace();
>         return null;
>     } catch (NullPointerException e) {
>         e.printStackTrace();
>         return null;
>     }
> 
>     try {
>         // audioFile is my original big videoFile. Index 0 is video, index 1 is audio  in Tracks.
>         // if you have a media file with only audio index can be 0. Be carefull
>         video.addTrack(audio.getTracks().get(1)); // 
>     } catch (Exception e) {
>         e.printStackTrace();
>     }
> 
>     Container out = new DefaultMp4Builder().build(video);
> 
>     FileOutputStream fos;
>     try {
>         fos = new FileOutputStream(outputFile.toString());
>     } catch (FileNotFoundException e) {
>         e.printStackTrace();
>         return null;
>     }
>     BufferedWritableFileByteChannel byteBufferByteChannel = new BufferedWritableFileByteChannel(fos);
>     try {
>         out.writeContainer(byteBufferByteChannel);
>         byteBufferByteChannel.close();
>         fos.close();
>     } catch (IOException e) {
>         e.printStackTrace();
>         return null;
>     }
>     return outputFile;
> }
> ```

**Comment by AnswerZhao** [comment]
@KAPLANDROID It doesn't work, how can you do it?

**Comment by kaplandroid** [comment]
@AnswerZhao  what is the error you are getting? Which line does not work?

**Comment by Berreis** [comment]
It worked. Thank you very much @KAPLANDROID

**Comment by iamnitesh** [comment]
i am getting error BufferedWritableFileByteChannel cannot be resolved to a type please give me some solution for that.. @KAPLANDROID

**Comment by kaplandroid** [comment]
@iamnitesh this is the class you need

 ```
private static class BufferedWritableFileByteChannel implements WritableByteChannel {
        private static final int BUFFER_CAPACITY = 1000000;

        private boolean isOpen = true;
        private final OutputStream outputStream;
        private final ByteBuffer byteBuffer;
        private final byte[] rawBuffer = new byte[BUFFER_CAPACITY];

        private BufferedWritableFileByteChannel(OutputStream outputStream) {
            this.outputStream = outputStream;
            this.byteBuffer = ByteBuffer.wrap(rawBuffer);
        }

        @Override
        public int write(ByteBuffer inputBuffer) throws IOException {
            int inputBytes = inputBuffer.remaining();

            if (inputBytes > byteBuffer.remaining()) {
                dumpToFile();
                byteBuffer.clear();

                if (inputBytes > byteBuffer.remaining()) {
                    throw new BufferOverflowException();
                }
            }

            byteBuffer.put(inputBuffer);

            return inputBytes;
        }

        @Override
        public boolean isOpen() {
            return isOpen;
        }

        @Override
        public void close() throws IOException {
            dumpToFile();
            isOpen = false;
        }

        private void dumpToFile() {
            try {
                outputStream.write(rawBuffer, 0, byteBuffer.position());
            } catch (IOException e) {
                throw new RuntimeException(e);
            }
        }
    }
```

**Comment by syc-pad** [comment]
**Big edit** :
Sorry, I plugged my brain off : it works pretty well, @KAPLANDROID  :) 👍

**Comment by northoutRavindra** [comment]
@KAPLANDROID Where should i add urs code in my application? Sorry i did not get urs solution please help me to short our from this problem..My requirement is that , i want to compress the video only...


Thanks

**Comment by northoutRavindra** [comment]
@HardikSuba @syc-pad @Berreis .. Did u get the solution, please help me to short out from this problem....My requirement is that i want to compress the video only...
Thanks

**Comment by kaplandroid** [comment]
@northoutRavindra  if you use my code sample you will only compress video track. Audio track remains same without any compression. Output files audio track is copy of original input files audio track
